### PR TITLE
Feature/update api version

### PR DESCRIPTION
**Pull request by rileymckenna**
## Description
* Bug fix in set connection
* Added Connection Identifier to Connection object.
* Updated to SDK 0.10.0-review
* Supports api-version 06-01-2019-preview for Microsoft.Peering

## Checklist

- [ x ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ X ] The title of the PR is clear and informative
- [ X ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ X ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ X ] If applicable, the changes made in the PR have proper test coverage
- [ X ] For public API changes to cmdlets:
    - [ X ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ X ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Peering/Peering.Test/ScenarioTests/PeeringCommon.ps1 [hunk 1]
```diff
@@ -75,15 +75,43 @@ function maxAdvertisedIpv6
...
+	Write-Debug("PeerName $asnPeer")
+	[string[]]$emails = "noc@$asnPeer.com","noc@$asnPeerName.com"
+	$phone = getAssetName "2342432433"
+	Write-Debug("Email: $emails; Phone $phone")
+	$created = New-AzPeerAsn -Name $asnPeerName -PeerName $asnPeer -PeerAsn $asnId -Email $emails -Phone $phone
+	Write-Debug("PeerName " + $created.Name)
+	$created.ValidationState = "Approved";
+	Write-Debug("ValidationState: " + $created.ValidationState)
+	Write-Debug("Sleep 2 seconds...")
+	Sleep(2)
```
Comment (cormacpayne) [inline src/Peering/Peering.Test/ScenarioTests/PeeringCommon.ps1]: @rileymckenna you should use `Wait-Seconds` here

File: src/Peering/Peering.Test/Peering.Test.csproj [hunk 2]
```diff
@@ -11,7 +11,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.Peering" Version="0.9.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.Peering" Version="0.10.0-preview" />
+    <PackageReference Include="Microsoft.Rest.ClientRuntime" Version="2.3.20" />
```
Comment (cormacpayne) [inline src/Peering/Peering.Test/Peering.Test.csproj]: @rileymckenna you can remove this dependency -- we have it defined already in a common `.targets` file that all `.csproj` files reference

File: src/Peering/Peering.Test/ScenarioTests/CreateNewExchangePeeringTests.cs [hunk 3]
```diff
@@ -59,15 +59,5 @@ public void TestNewExchangePeeringPipe()
         {
             TestController.NewInstance.RunPowerShellTest(this.logger, "Test-NewExchangePeeringPipe");
         }
-
-        /// <summary>
-        /// The test new exchange peering pipe two connections.
-        /// </summary>
-        [Fact]
-        [Trait(Category.AcceptanceType, Category.CheckIn)]
-        public void TestNewExchangePeeringPipeTwoConnections()
```
Comment (cormacpayne) [inline src/Peering/Peering.Test/ScenarioTests/CreateNewExchangePeeringTests.cs]: @rileymckenna any reason this test is being removed?

File: src/Peering/Peering.Test/ScenarioTests/PeeringCommon.ps1 [hunk 4]
```diff
@@ -75,15 +75,43 @@ function maxAdvertisedIpv6
 function getHash
 {
 $hash = getPeeringVariable "Hash" $ipGenerator.BuildHash()
+Write-Debug("The hash $hash");
```
Comment (cormacpayne) [inline src/Peering/Peering.Test/ScenarioTests/PeeringCommon.ps1]: @rileymckenna in this file, it should be `Write-Debug "Message"` instead of `Write-Debug("Message")`

File: src/Peering/Peering.Test/ScenarioTests/ScenarioExchangePeeringTests.cs [hunk 5]
```diff
@@ -57,26 +57,6 @@ public void TestConvertLegacyToExchange()
             TestController.NewInstance.RunPowerShellTest(this.logger, "Test-ConvertLegacyToExchange");
         }
 
-        /// <summary>
-        /// The test update exchange i pv 4 on resource id.
-        /// </summary>
-        [Fact]
-        [Trait(Category.AcceptanceType, Category.CheckIn)]
```
Comment (cormacpayne) [inline src/Peering/Peering.Test/ScenarioTests/ScenarioExchangePeeringTests.cs]: @rileymckenna any reason the tests in this file are being removed?

File: src/Peering/Peering.Test/ScenarioTests/ScenarioExchangePeeringTests.ps1 [hunk 6]
```diff
@@ -56,87 +64,83 @@ function Test-ConvertLegacyToExchange
...
 	$oldpeering = $peering
-	$peering.Connections[0] = $peering.Connections[0] |  Set-AzPeeringExchangeConnectionObject -PeerSessionIPv4Address $newipv4 -MaxPrefixesAdvertisedIPv4 $maxv4
-	$peering.Connections[1] = $peering.Connections[1] |  Set-AzPeeringExchangeConnectionObject -PeerSessionIPv4Address $newipv42 -MaxPrefixesAdvertisedIPv4 $maxv42
-	$update = Update-AzPeering -ResourceId $peering.Id $peering.Connections
+	$tags = New-Object 'system.collections.generic.dictionary[string,string]'
+	$tags["tfs_234234"] = "Active";
+	$peering.Tags = $tags;
+	$peering.Connections[0] = $peering.Connections[0] |  Set-AzPeeringExchangeConnectionObject -PeerSessionIPv4Address $newipv4
+	$peering.Connections[1] = $peering.Connections[1] |  Set-AzPeeringExchangeConnectionObject -PeerSessionIPv4Address $newipv42
+	Write-Debug("ResourceId:"+$peering.Id)
```
Comment (cormacpayne) [inline src/Peering/Peering.Test/ScenarioTests/ScenarioExchangePeeringTests.ps1]: @rileymckenna same comment about changing this to `Write-Debug "Message"`

File: src/Peering/Peering.Test/ScenarioTests/ScenarioExchangePeeringTests.ps1 [hunk 7]
```diff
@@ -56,87 +64,83 @@ function Test-ConvertLegacyToExchange
...
+	$newipv6 = getPeeringVariable "newIpv6" (changeIp "$ipv6/128" $true 14 $false)
+	$ipv62 = $peering.Connections[1].BgpSession.PeerSessionIPv6Address
+	$newipv62 = getPeeringVariable "newIpv62" (changeIp "$ipv6/128" $true 16 $false)
 	$oldpeering = $peering
-	$peering.Connections[0] = $peering.Connections[0] |  Set-AzPeeringExchangeConnectionObject -PeerSessionIPv4Address $newipv4 -MaxPrefixesAdvertisedIPv4 $maxv4
-	$peering.Connections[1] = $peering.Connections[1] |  Set-AzPeeringExchangeConnectionObject -PeerSessionIPv4Address $newipv42 -MaxPrefixesAdvertisedIPv4 $maxv42
-	$update = $peering | Update-AzPeering 
+	$peering.Connections[0] = $peering.Connections[0] |  Set-AzPeeringExchangeConnectionObject -PeerSessionIPv6Address $newipv6 
+	$peering.Connections[1] = $peering.Connections[1] |  Set-AzPeeringExchangeConnectionObject -PeerSessionIPv6Address $newipv62
+	Write-Debug($peering.Name)
```
Comment (cormacpayne) [inline src/Peering/Peering.Test/ScenarioTests/ScenarioExchangePeeringTests.ps1]: @rileymckenna same comment here about `Write-Debug "Message"`

File: src/Peering/Peering.Test/ScenarioTests/SetPeeringTests.cs [hunk 8]
```diff
@@ -51,26 +51,6 @@ public void TestGetAndSetUseForPeeringService()
             TestController.NewInstance.RunPowerShellTest(this.logger, "Test-GetAndSetUseForPeeringService");
         }
 
-        /// <summary>
-        /// The test set new ip.
-        /// </summary>
-        [Fact]
-        [Trait(Category.AcceptanceType, Category.CheckIn)]
```
Comment (cormacpayne) [inline src/Peering/Peering.Test/ScenarioTests/SetPeeringTests.cs]: @rileymckenna any reason the tests in this file are being removed?

File: src/Peering/Peering/ChangeLog.md [hunk 9]
```diff
@@ -19,6 +19,12 @@
 -->
 ## Upcoming Release
 
+## Version 0.2.0
```
Comment (cormacpayne) [inline src/Peering/Peering/ChangeLog.md]: @rileymckenna you can remove this version header and just put everything under `## Upcoming Release` -- we will do our own versioning before we release the module

File: src/Peering/Peering/Peering.csproj [hunk 10]
```diff
@@ -12,7 +12,8 @@
 
   <ItemGroup>
     <PackageReference Include="AutoMapper" Version="6.2.2" />
-    <PackageReference Include="Microsoft.Azure.Management.Peering" Version="0.9.0-preview" />
+    <PackageReference Include="Microsoft.Rest.ClientRuntime" Version="2.3.20" />
```
Comment (cormacpayne) [inline src/Peering/Peering/Peering.csproj]: @rileymckenna same comment about removing this dependency since it's already defined in the `.targets` file

File: src/Peering/Peering/help/Get-AzLegacyPeering.md [hunk 11]
```diff
@@ -58,7 +58,7 @@ Gets the legacy peering for Seattle
 The credentials, account, tenant, and subscription used for communication with Azure.
 
 ```yaml
-Type: Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer
```
Comment (cormacpayne) [inline src/Peering/Peering/help/Get-AzLegacyPeering.md]: @rileymckenna please regenerate the help content using the `-UseFullTypeName` switch parameter when running [`Update-MarkdownHelpModule`](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module) cmdlet

File: src/Peering/Peering.Test/ScenarioTests/SetPeeringTests.cs (hunk 8 above)
Comment (rileymckenna) [reply src/Peering/Peering.Test/ScenarioTests/SetPeeringTests.cs]: These were duplicate tests. or they aren't supported by the back end right now.

File: src/Peering/Peering.Test/ScenarioTests/ScenarioExchangePeeringTests.cs (hunk 5 above)
Comment (rileymckenna) [reply src/Peering/Peering.Test/ScenarioTests/ScenarioExchangePeeringTests.cs]: These were duplicate tests. or they aren't supported by the back end right now.

File: src/Peering/Peering.Test/ScenarioTests/CreateNewExchangePeeringTests.cs (hunk 3 above)
Comment (rileymckenna) [reply src/Peering/Peering.Test/ScenarioTests/CreateNewExchangePeeringTests.cs]: These were duplicate tests. or they aren't supported by the back end right now.

Comment (cormacpayne) [reply src/Peering/Peering.Test/ScenarioTests/CreateNewExchangePeeringTests.cs]: Just to confirm, when you say that a test isn't supported in the back end right now, does that mean it previously worked and you were able to record the test previously, but now does not work when trying to record the test against the live server?

**Comment by rileymckenna** [comment]
So the missing tests were duplicates or were no longer supported by the back end.
